### ISSUE TITLE
[Feature Request] Podman support

### ISSUE THREAD

**Issue by deadmeu** [nilsherzig/LLocalSearch#37]
I tried to start this with `podman compose up` and got several errors like the following:

`Error: short-name "nilsherzig/llocalsearch-backend:latest" did not resolve to an alias and no unqualified-search registries are defined in "/etc/containers/registries.conf"`

I think this can be resolved by explicitly including the host in the container URLs?

I know you're using docker, but it would be nice if podman was supported as well.

**Comment by nilsherzig** [comment]
> So after specifying `docker.io` as the image hosts I can get the frontend up, but I had some DNS issues. I resolved these DNS issues by installing the `aardvark-dns` package on Arch.
> 
> Now I'm having issue pulling a model:
> 
> ```
> 2024/04/04 12:43:52 WARN Model does not exist, pulling it model=all-minilm:v2
> 2024/04/04 12:43:52 ERROR Model does not exist and could not be pulled model=all-minilm:v2 error="Post \"http://host.docker.internal:11434/api/pull\": dial tcp 10.0.0.3:11434: connect: connection refused"
> ```
> 
> It's trying to call something on `docker.internal`?

yes by default https://github.com/nilsherzig/LLocalSearch/blob/d3d6ed627baa4398eb93a77fa755f6f9f18d2371/docker-compose.yaml#L5

**Comment by nilsherzig** [comment]
> Actually, I think we can use `172.17.0.1` instead, but that isn't pointing to ollama properly.

I don't know if podman uses the defined private container networks (check "networks:" in the compose file). But if thats the case, "localhost" on the backend isn't your host systems "localhost"

**Comment by nilsherzig** [comment]
![image](https://github.com/nilsherzig/LLocalSearch/assets/72463901/43f45321-cfba-43b8-a9b5-2eb5c76791e8)

as you can see the container has its own loopback device / locahost. That's why we use host.docker.internal to connect to the host.
